_posts/2020-03-06-introducing-mirrormaker2.md

```diff
@@ -0,0 +1,268 @@
+metadata:
+  name: my-mirror-maker2
+spec:
+  version: {DefaultKafkaVersion}
+  replicas: 3
+  connectCluster: "my-cluster-target"
+    clusters:
+    - alias: "my-cluster-source"
+    authentication:
+      certificateAndKey:
+        certificate: source.crt
+        key: source.key
+        secretName: my-user-source
+      type: tls
+    bootstrapServers: my-cluster-source-kafka-bootstrap:9092
+    tls:
+      trustedCertificates:
+      - certificate: ca.crt
+        secretName: my-cluster-source-cluster-ca-cert
+  - alias: "my-cluster-target"
+    authentication:
+      certificateAndKey:
+        certificate: target.crt
+        key: target.key
+        secretName: my-user-target
+      type: tls
+    bootstrapServers: my-cluster-target-kafka-bootstrap:9092
+    config:
+      config.storage.replication.factor: 1
+      offset.storage.replication.factor: 1
+      status.storage.replication.factor: 1
+    tls:
+      trustedCertificates:
+      - certificate: ca.crt
+        secretName: my-cluster-target-cluster-ca-cert
+```
+
+People can be confused by the idea of _replicas_ and _replication_ through MirrorMaker,
+that's why you'll see replication between clusters often referred to as _mirroring_.
+Just thought I'd bring that up in case I've been inconsistent in this post.
+
+If you don't want to leave the defaults, you can also include configuration for the **_MirrorMaker 2.0 connectors_** and related **_internal topics_**.
+
+The `config` overrides the default configuration options, so here we alter the replication factors for the **_internal topics_**:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker2
+spec:
+# ...
+  mirrors:
+  # ...
+  sourceConnector:
+    config:
+      replication.factor: 1
+      offset-syncs.topic.replication.factor: 1
+  heartbeatConnector:
+    config:
+      heartbeats.topic.replication.factor: 1
+  checkpointConnector:
+    config:
+      checkpoints.topic.replication.factor: 1
+```
+
+You can see the full `spec` options in the [KafkaMirrorMaker2 schema reference](https://strimzi.io/docs/master/#type-KafkaMirrorMaker2-reference).
+
+If, at this point, you're wondering what happens if you're using the old version of MirrorMaker,
+it's still supported.
+And there are also plans to have a _legacy mode_ in MirrorMaker 2.0 that creates topics without the cluster prefix,
+and doesn't do the topic configuration mirroring.
+Basically, turning off the main differences between the original and the new version of MirrorMaker.
+
+### Embrace change
+
+Apache understood that MirrorMaker was due an overhaul [[KIP-382: MirrorMaker 2.0](https://cwiki.apache.org/confluence/display/KAFKA/KIP-382%3A+MirrorMaker+2.0)].
+They identified the key issues with using the original MirrorMaker -- manual topic configuration, the lack of support for _active/active_ replication, and the inability to track offsets -- and eradicated them with MirrorMaker 2.0.
```

Comment:
Can we reword this to avoid "they", that kind of supposes some kind of "them" and "us" split which is not the case. "Key usage issues with the original MorrirMaker... were identified" for example.

Reply:
Good idea. Changed.
Paolo mentioned using _Apache Kafka community_ as well.

> _Key issues with using the original MirrorMaker were identified -- manual topic configuration, the lack of support for _active/active_ replication, and the inability to track offsets -- and eradicated with MirrorMaker 2.0._